packages/react/src/use-feature-flag.ts

```diff
@@ -0,0 +1,47 @@
+import { Client, EvaluationDetails, FlagValue, ProviderEvents, ResolutionDetails } from '@openfeature/web-sdk';
+import { useEffect, useState } from 'react';
+import { useOpenFeatureClient } from './provider';
+
+export function useFeatureFlag<T extends FlagValue>(flagKey: string, defaultValue: T): T {
+  const [flagEvaluationDetails, setFlagEvaluationDetails] = useState<ResolutionDetails<T> | null>(null);
```

Comment:
🚨 this approach may go poorly 🚨

I took a very similar tack in an internal implementation  a `useFeatureFlag` at $DAY_JOB$, and it bit us pretty hard. Trying to maintain your own copy of the flag evaluation state separate from the OF client leads to some very confusing race conditions due to how React rendering and hooks work. 

It is MUCH better to simply call `getFlag` each time the hook is evaluated.

Reply:
That makes sense to me.

Reply:
And maintain/manage any sort of caches within the OpenFeature provider?

Reply:
> And maintain/manage any sort of caches within the OpenFeature provider?

I won't speak for @moredip , but that's how I took it... and I think it's consistent with all our paradigms and thinking so far. Maintaining state in 2 places is always a bad idea:tm:

Reply:
> It is MUCH better to simply call getFlag each time the hook is evaluated.

Ya, +1 on that.
Would we then just add something like a "stale" state that we set when ProviderEvents.Ready and ProviderEvents.ConfigurationChanged occur?  

Could be something like this sketch: 

```ts
 const [, updateFlag] = React.useState();
 const reevaluateFlag = React.useCallback(() => updateFlag({}), []);

  useEffect(() => {
    // adding handlers here means that changes are immediately reflected in the UI when flags change
    client.addHandler(ProviderEvents.Ready, reevaluateFlag);
    client.addHandler(ProviderEvents.ConfigurationChanged, reevaluateFlag);
    return () => {
      // be sure to cleanup the handlers
      client.removeHandler(ProviderEvents.Ready, reevaluateFlag);
      client.removeHandler(ProviderEvents.ConfigurationChanged, reevaluateFlag);
    };
  }, [flagKey, defaultValue, client]);

  return getFlag(client, flagKey, defaultValue, setFlagEvaluationDetails)

```